wagtail/admin/templatetags/wagtailadmin_tags.py

```diff
@@ -708,6 +708,21 @@ def admin_theme_classname(context):
     return f"w-theme-{theme_name} w-density-{density_name} w-contrast-{contrast_name}"
 
 
+@register.simple_tag(takes_context=True)
+def admin_theme_color_scheme(context):
+    """
+    Retrieves the color-scheme of the active theme for current user.
+    """
+    user = context["request"].user
+    theme_name = getattr(user, "wagtail_userprofile", None)
+    theme_name = theme_name.theme if theme_name else "light"
```

Comment:
It's not correct to store the UserProfile object in a `theme_name` variable, because it's not the theme name.

```suggestion
    profile = getattr(user, "wagtail_userprofile", None)
    theme_name = profile.theme if profile else "system"
```